NUI-5753: standarize-and-document-how-we-mark-deprecations (#374)
solarwinds/nova: docs/DEPRECATIONS_AND_BREAKING_CHANGES.md

1. API breaking changes can be expected in conjunction with a Nova major version increase. Read the release notes carefully to look for the deprecated items that are set for removal. See the [CHANGELOG.md](./CHANGELOG.md) for details;
2. Everything that has had a `@deprecated` decorator for a full major version will likely to be removed with the next major version increase. This will be also reflected in a migration guide.
3. If a deprecated entity has been in this status for less than a full major version, it will be carried over into the next major version, and it will be removed in the one after it.

## Our Deprecation Guidelines

> 1. Create a new [ticket in github](https://github.com/solarwinds/nova/issues/new?assignees=&labels=&template=deprecation_report.md&title=) and fill in all the required information.

Reviewer: Suggested tweak:

```suggestion
1. Create a new [breaking changed](https://github.com/solarwinds/nova/issues/new?assignees=&labels=&template=breaking_change_task.md&title=) ticket in github and fill in all the required information.
```

> 2. Inside a jsdoc style comment above the depreacted item

Reviewer: ```suggestion
2. Inside a jsdoc style comment above the deprecated item
```

>     - use <code>@deprecated</code>

Reviewer: ```suggestion
    - use the <code>@deprecated</code> tag
```

>     - state the version in which the deprecation occurred

Reviewer: ```suggestion
    - state the major version in which the deprecation occurred
```

>     - Include the recommended non-deprecated replacement

Reviewer: ```suggestion
    - Include the recommended non-deprecated replacement that consumers can use instead
```

>     - and the github id for the ticket you just created.

Reviewer: ```suggestion
    - Include the github id for the ticket you just created
```

1. API breaking changes can be expected in conjunction with a Nova major version increase. Read the release notes carefully to look for the deprecated items that are set for removal. See the [CHANGELOG.md](./CHANGELOG.md) for details;
2. Everything that has had a `@deprecated` decorator for a full major version will likely to be removed with the next major version increase. This will be also reflected in a migration guide.
3. If a deprecated entity has been in this status for less than a full major version, it will be carried over into the next major version, and it will be removed in the one after it.

> ## Our Deprecation Guidelines

Reviewer: ```suggestion
## Steps for Introducing a Deprecation
```
